lib/helpers/handle-accept-request.js

```diff
@@ -1,31 +1,74 @@
 'use strict';
 
+var _ = require('lodash');
+
+var SpaResponseHandler = require('./spa-handler');
+
 /**
- * Handles an HTTP Accept request.
+ * Determines which handler should be used to fulfill a response, given the
+ * Accept header of the request and the content types that are allowed by the
+ * `stormpath.web.produces` configuration.  Also handles the serving of the SPA
+ * root page, if needed.
  *
  * @method
  * @private
  *
  * @param {Object} req - HTTP request.
  * @param {Object} res - HTTP response.
- * @param {Object} handlers - Object where the key is the content type's available to accept.
- * @param {function} fallbackHandler - Handler to call when no content type was accepted.
+ * @param {Object} handlers - Object where the keys are content types and the
+ * functions are handlers that will be called, if needed, to fulfill the
+ * response as that type.
+ * @param {Function} fallbackHandler - Handler to call when an acceptable
+ * content type could not be resolved.
  */
 function handleAcceptRequest(req, res, handlers, fallbackHandler) {
   var config = req.app.get('stormpathConfig');
-  var accepted = req.accepts(config.web.produces);
 
-  if (!accepted || !(accepted in handlers)) {
-    return fallbackHandler();
-  }
+  // accepted is an ordered list of preferred types, as specified by the request
+
+  var accepted = req.accepts();
+
+  var produces = config.web.produces;
+
+  /*
+    Our default response is HTML, if the client does not specify something more
+    specific.  As such, map the wildcard type to html.
+  */
+
+  accepted = accepted.map(function (contentType) {
+    return contentType === '*/*' ? 'text/html' : contentType;
+  });
+
+  // Of the accepted types, find the ones that are allowed by the configuration
 
-  // If this is a text/html response, then check if we have SPA-mode enabled.
-  // If it is enabled, then return the view for it.
-  if (accepted === 'text/html' && config.web.spa.enabled) {
-    return res.sendFile(config.web.spa.view);
+  var allowedResponseTypes = _.intersection(produces, accepted);
+
+  /*
+    Of the allowed response types, find the first handler that matches. But
+    always override with the SPA handler if SPA is enabled.
+  */
+
+  var handler;
+
+  allowedResponseTypes.forEach(function (contentType) {
```

Comment:
You can use `.some()` or `.find()` here instead and then return `true` to abort iteration.